docs/spec/json-web-signatures.md

```diff
@@ -19,25 +19,48 @@ Here is an example of a decoded JWS header:
 }
 ```
 
+::: warning
+  It is recommended that `kid` be a DID URL. If it is not, method implementers might need to rely on additional context to uniquely identify the associated verificationMethod. 
+:::
+
 ### Verifying
 
-If `kid` is a DID URL, the verification key MAY be resolved as follows:
+Regardless of which proof purpose a verificationMethod is associated with, the process of verifying a JWS linked to a DID is the same.
 
-1. Resolve the `kid` value using a resolver.
-2. iterate the `operations` and `recovery` verificationMethods, until a verificationMethod with `id` equal to `kid` is found.
-3. A [Recover Operation](https://identity.foundation/sidetree/spec/#recover) is only valid when signed by a key listed in `recovery`. Other operations are only valid when signed with keys listed in `operations`. If the key is not present in the expected collection, verification fails, and the operation is considered invalid. If the key is present in the expected collection proceed.
-3. Convert the discovered verificationMethod to JWK if necessary.
-4. Perform [JWS Verification](https://tools.ietf.org/html/rfc7515#section-5.2) using the JWK.
-5. The operation is considered valid if key material was present in the correct collection and the signature is valid.
+The JWS header is parsed and a `kid` is extracted.
 
+1. Iterate the verificationMethods, until a verificationMethod with `id` equal to `kid` is found.
+2. Convert the discovered verificationMethod to JWK if necessary.
+3. Perform [JWS Verification](https://tools.ietf.org/html/rfc7515#section-5.2) using the JWK.
 
-If `kid` is not a DID URL, the verification key MAY be resolved as follows:
+#### Operation Verification
 
-1. Iterate an internal representation of keys registered for use with the DID until a jwk with `kid` equal to `kid` is found.
-2. Ensure that the `usage` property of the internal key represenation is `recovery` if the operation is a recovery operation or `ops` if the operation is not Recovery or Update.
-3. Perform [JWS Verification](https://tools.ietf.org/html/rfc7515#section-5.2) using the JWK.
-4. The operation is considered valid if key material was present with the correct usage and the signature is valid.
+Sidetree operations are considerd valid when the JWS can be verified, and where the key used is associated with the correct proof purpose.
+
+`operations`, `recovery` are proof purposes for verifying sidetree operations which sidetree DID Methods MUST support, but which MAY NOT be exposed externally via the DID Document or Resolver Method Meta Data.
```

Comment:
@csuwildcat I think its still true that all Sidetree DID Methods MUST support operations and recovery... I think what you are saying is that these proof purposes MAY NOT be exposed via either the did document or resolver meta data.

Reply:
To clarify: I am definitely saying all Sidetree methods need to somehow track keys in relation to those two purposes, because they need to know what keys do what things, but simply asking that we either don't forcibly project out op/recovery-only keys into the doc, or at least make it optional and caution that if you do project these keys, a wallet implementer must be hyper vigilant in the DID Doc-referenced invocations they execute in their business logic, because leaving a 'hanging chad' could mean the wallet inadvertently interacts with a key and an external party that was not intended.

If this text means: "You need to have a _concept_ of these purposes, such that an op can be processed against the right key, but DO NOT project the keys into the doc unless they are explicitly marked as dual-use keys the user wants to put in, say, `authentication`", then yes, I am totally fine with that. I was not arguing that it had to be a MAY NOT on the latter part of that language, because I wanted to be sensitive to others who view the attack surface/vectors differently than I do. Still fine just leaving that a MAY, if it makes you more comfortable for Element.

Reply:
@csuwildcat can you review again, and either approve or request change.